crates/bevy_ecs/src/observer/runner.rs

```diff
@@ -226,6 +226,21 @@ impl Observer {
         }
     }
 
+    /// Creates a new [`Observer`], which defaults to a "global" observer. This means it will run whenever the event `E` is triggered
+    /// for _any_ entity (or no entity). The [`Observer`] will have a [`Name`] derived from the system.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the given system is an exclusive system.
+    pub fn new_with_derived_name<E: Event, B: Bundle, M, I: IntoObserverSystem<E, B, M>>(
+        system: I,
+    ) -> (Name, Observer) {
+        (
+            Name::new(format!("Observer {}", type_name_of_val(&system))),
+            Self::new(system),
+        )
```

Comment:
Do you want to use `System::name` here?  Using `type_name` directly means it won't see any names set using `FunctionSystem::with_name`.  

```suggestion
        let observer = Self::new(system);
        (
            Name::new(format!("Observer {}", observer.system.name())),
            observer,
        )
```

Since that's available from the trait object, you could even rework it as `fn with_derived_name(self) -> impl Bundle` so that you can do `Observer::new(observer).with_entity(self.entity).with_derived_name()` as a single expression below.

Reply:
That's a thing?

Reply:
new attempt was done with this in mind